server/api/org/update/controller.js

```diff
@@ -0,0 +1,98 @@
+import Joi from 'joi';
+import Boom from 'boom';
+import compose from 'koa-compose';
+import packJSONRPC from '../../../middleware/packJSONRPC';
+import { validateRequest } from '../../../middleware/validation';
+import {
+  decorateSession,
+  isUserAuthenticated,
+  decorateUserPermissions,
+  findUserPermissionIndex,
+} from '../../../middleware/auth';
+import updateOrg, { checkOrgExists } from './service';
+import { AuthorizationError, OrgNotFoundError } from '../../../constants/errors';
+
+export const validationSchema = {
+  body: {
+    id: Joi.string()
+      .length(24)
+      .hex()
+      .required(),
+    name: Joi.string()
```

Comment:
He's an arguably better way to validate `name` as required when `slug` does not exist...

```javascript
name: Joi.string()
  .trim()
  .min(3)
  .max(100)
  .when('slug', {
    is: Joi.exist(),
    then: Joi.optional(),
    otherwise: Joi.required(),
  })
```

Reply:
This one unfortunatelly starts a recursive loop because we are using a circular dependancy (as also [stated in this issue](https://github.com/hapijs/joi/issues/588))

This is the error thrown.
"item added into group slug created a dependencies error(slug)

I believe this might be the reason you also had this kind of checks (in another controller) instead of adding a more complex Joi check (i didn't quite found a clear fix. Will need to check better if we want to delegate this issue to Joi).

Please advise.

Reply:
Agreed. Thanks for looking into this.